### PR TITLE
Update wgpu to 0.18

### PR DESCRIPTION
**Pull request by sagudev**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes in WPT

### CODE REVIEW

**Comment by sagudev** [comment]
I have try run in my fork, then I will update expectations and retest and hopefully land.

**Review by mrobinson — APPROVED** [review]

File: servo-tidy.toml [hunk 1]
```diff
@@ -54,6 +54,8 @@ packages = [
     "foreign-types-shared",
     "metal",
     "paste",
+    "hashbrown",
+    "indexmap",
```
[the remark this answers is not in the document]
Comment (sagudev) [reply servo-tidy.toml:58]:
```
./Cargo.lock:1: duplicate versions for package `hashbrown`
        The following packages depend on version 0.12.3 from 'crates.io':
                gpu-descriptor 0.2.3
                indexmap 1.9.3
        The following packages depend on version 0.14.2 from 'crates.io':
                indexmap 2.1.0
./Cargo.lock:1: duplicate versions for package `indexmap`
        The following packages depend on version 1.9.3 from 'crates.io':
                cbindgen 0.26.0
                clap 3.2.25
                h2 0.3.21
                malloc_size_of 0.0.1
                script 0.0.1
                style 0.0.1
        The following packages depend on version 2.1.0 from 'crates.io':
                naga 0.14.0
```

**Comment by sagudev** [comment]
There are some expectation failures also on master: https://github.com/sagudev/servo/actions/runs/6747438130/job/18344589176

I am also getting
urllib.error.HTTPError: HTTP Error 413: Request Entity Too Large
at the end of invocation: https://github.com/sagudev/servo/actions/runs/6747438130/job/18344589176#step:10:13446 when intermittent filtering is done and this is very annoying when trying to use expectations from CI because I am not getting any.

**Comment by sagudev** [comment]
Expectation failures are from debug assertions (some are crashes, some are timeouts).

**Comment by sagudev** [comment]
This needs https://github.com/servo/servo/pull/30694 so we can land with confidence.

**Comment by sagudev** [comment]
replaced with https://github.com/servo/servo/pull/30926

**Pull request closed without merging**